hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/impl/BurnPrecompile.java

```diff
@@ -89,10 +81,14 @@ public BurnPrecompile(
 
     @Override
     public TransactionBody.Builder body(final Bytes input, final UnaryOperator<byte[]> aliasResolver) {
-        burnOp = switch (functionId) {
-            case AbiConstants.ABI_ID_BURN_TOKEN -> decodeBurn(input);
-            case AbiConstants.ABI_ID_BURN_TOKEN_V2 -> decodeBurnV2(input);
-            default -> null;};
+        final var burnAbi =
+                switch (functionId) {
+                    case AbiConstants.ABI_ID_BURN_TOKEN -> SystemContractAbis.BURN_TOKEN_V1;
+                    case AbiConstants.ABI_ID_BURN_TOKEN_V2 -> SystemContractAbis.BURN_TOKEN_V2;
+                    default -> throw new IllegalArgumentException("invalid selector to burn precompile");
+                };
+        burnOp = getBurnWrapper(input, burnAbi);
+        if (burnOp == null) throw new IllegalArgumentException("unable to create burn wrapper from decoded input");
```

Comment:
Q: where is it possible for `burnOp` to be null?
Won't all cases of `getBurnWrapper()` call either the FT or NFT case which will be non null values.

If not possible what can we check for instead here?

Reply:
I think the only issue was the BurnWrapper factories weren't marked `@NonNull` as they should have been.  I fixed it in the wrong place and in the wrong way.  Thanks!